lib/src/base_client.dart

```diff
@@ -55,16 +56,15 @@ abstract class BaseClient implements Client {
 
   Future<Response> send(Request request);
 
+  void close() {}
+
   /// Throws an error if [response] is not successful.
-  void _checkResponseSuccess(url, Response response) {
+  static void _checkResponseSuccess(url, Response response) {
```

Comment:
I don't generally like making helper methods static unless there's a good reason to do so.

Reply:
Using `static` is a bit more internally consistent with the rest of the code.

Reply:
How so? The only other places I see `static` are where it's necessary to use the methods from constructors.

Reply:
Sorry you're correct I missed where the other ones were being called.